components/admin_console/team_channel_settings/team/details/team_details.tsx

```diff
@@ -339,15 +339,16 @@ export default class TeamDetails extends React.PureComponent<Props, State> {
     addUsersToAdd = (users: UserProfile[]) => {
         let {usersToRemoveCount} = this.state;
         const {usersToAdd, usersToRemove} = this.state;
+        const usersToAddCopy = cloneDeep(usersToAdd);
         users.forEach((user) => {
             if (usersToRemove[user.id]?.id === user.id) {
                 delete usersToRemove[user.id];
                 usersToRemoveCount -= 1;
```

Comment:
We are also modifying state directly here, should we also fix this?

Reply:
Thanks for raising this Ben.

So this was inspired from https://stackoverflow.com/a/55567414 , and I could see that other appends i.e to `usersToRemove` work just fine.

Surely this doesn't fix the root cause of the issue here, I think. I'll keep looking.